In [6]:
!pip install openpyxl

     -------------------------------------- 242.1/242.1 kB 7.5 MB/s eta 0:00:00


# START FROM HERE

In [26]:
#SCRAPPING SINGLE ITEM FROM THE WEBSITE

import requests
from bs4 import BeautifulSoup

class PriceTracker:
    def __init__(self, url):
        self.url=url
        self.user_agent={"User-Agent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"}
        self.response=requests.get(self.url, headers=self.user_agent).text
        self.soup= BeautifulSoup(self.response,"lxml")
    
    def product_title(self):
        title=self.soup.find("h1",{"class":"product-meta__title heading h1"})
        if title != None:
            return title.text.strip()
        return "tag not found"
    
    def product_price(self):
        price=self.soup.find("span",{"class":"price price--highlight"})
        if price != None:
#             price=self.soup.find()
            return price.text.strip()
        return "tag not found"

    def product_description(self):
        desc=self.soup.find("div",{"class":"rte text--pull"})
        if desc != None:
            return desc.text.strip()
        return "tag not found"
    
    
    
    
    
li= PriceTracker(url="https://werrv.ca/products/kvh-hr24-hd-dvr-receiver-110v-ac-f-directv-w-rf-ir-remote-control-72-0900-hr24")
print(li.product_title())
print(li.product_price())
print(li.product_description())

KVH HR24 HD/DVR Receiver - 110V AC f/DIRECTV w/RF/IR Remote Control - *Remanufactured [72-0900-HR24]
Sale price$513.00
HR24 HD/DVR Receiver - 110V AC for DIRECTV with RF/IR Remote Control - *RemanufacturedKVH Satellite Receiver with RF remote can be mounted in cabinets and hidden away located. Ideal for marine installations. The HR24 for DIRECTV is an HD DVR satellite receiver that ensures watching and recording television is convenient and seamless. Record two programs at once, stream on-demand video, watch Youtube videos and listen to Pandora, all from the same device. Sleek design with a touch sensitive panel, it comes with a self-programming remote and operates quietly. Energy-star certified. Features:
500GB hard drive 400hrs SD
100hrs HD record time
2.8" H x 15.2" W x 11.8" D
110 VAC
Includes RF Remote control
HDMI
S-video
PLEASE CONTACT KVH FOR RECEIVER ACTIVATION: directv@kvh.com


# METHOD 2 WITHOUT USING CLASS

In [27]:
import requests
from bs4 import BeautifulSoup

url = "https://werrv.ca/products/kvh-hr24-hd-dvr-receiver-110v-ac-f-directv-w-rf-ir-remote-control-72-0900-hr24"

response = requests.get(url)
soup = BeautifulSoup(response.content, "html.parser")

# Extract the product title
product_title = soup.find("h1",{"class":"product-meta__title heading h1"}).text
print("Product Title:", product_title)

# Extract the product price
product_price = soup.find("span",{"class":"price price--highlight"}).text
print("Product Price:", product_price)

Product Title: KVH HR24 HD/DVR Receiver - 110V AC f/DIRECTV w/RF/IR Remote Control - *Remanufactured [72-0900-HR24]
Product Price: 
Sale price$513.00


# NOW WE GONNA TRY TAKING ALL THE PRODUCTS FROM THE PAGE 

In [31]:
import requests
from bs4 import BeautifulSoup

url = "https://werrv.ca/collections/automotive-rv?sort_by=best-selling&filter.v.availability=1"

response = requests.get(url)

soup = BeautifulSoup(response.text, 'html.parser')

# Find all the products on the page
products = soup.find_all("div", {"class": "product-item__info-inner"})

data=[]
for product in products:
    title = product.find("a", {"class": "product-item__title text--strong link"}).text
    price = product.find("div", {"class": "product-item__price-list price-list"}).text
    data.append([title,price])
#     description = product.find("div", {"class": "product-card__description"}).text
#     print("Title: ", title)
#     print("Price: ", price)
#     print("Description: ", description)
#     print("\n")


In [33]:
df = pd.DataFrame(data, columns=['Title', 'Price'])

# Export the dataframe to an excel sheet
df.to_excel("werrv.xlsx", index=False)

In [34]:
#using this we can now collect all the products from that parrticular page

# NOW WE GONNA DO FOR ALL PAGE

In [35]:
import requests
from bs4 import BeautifulSoup
import pandas as pd

# Start URL
url = "https://werrv.ca/collections/automotive-rv?sort_by=best-selling&filter.v.availability=1"

#initialize empty list
data = []

while url:
    # Send a GET request to the URL
    response = requests.get(url)

    # Create a BeautifulSoup object to parse the HTML content
    soup = BeautifulSoup(response.text, 'html.parser')

    # Find all the products on the page
    products = soup.find_all("div", {"class": "product-item__info-inner"})

    for product in products:
        title = product.find("a", {"class": "product-item__title text--strong link"}).text
        price = product.find("div", {"class": "product-item__price-list price-list"}).text
        data.append([title,price])
    
    # Find the next page button
    next_button = soup.find("a", {"class": "pagination__next link"})

    # If there is a next page button, update the URL
    if next_button:
        url = "https://werrv.ca" + next_button['href']
    else:
        # If there is no next page button, break out of the loop
        url = None
        
# Create a dataframe from the scraped data
df = pd.DataFrame(data, columns=['Title', 'Price'])

# Export the dataframe to an excel sheet
df.to_excel("FINAL.xlsx", index=False)


In [36]:
# https://werrv.ca/collections/automotive-rv?sort_by=best-selling&filter.v.availability=1
# https://werrv.ca/collections/automotive-rv?sort_by=best-selling&filter.v.availability=1&page=2
# https://werrv.ca/collections/automotive-rv?sort_by=best-selling&filter.v.availability=1&page=3

In [25]:
# there are sub categories in the each list so try to put all the categories inside the for loop and try to pull out all the
# data in a single class

#  we having a google search for loop ready, we need to to search the keywords from the werrv and need to search the same 
# in the loop to get a google search

In [24]:
# introduction
# werv
# problem statement
# planned techniues gonna use 
# goal
# resources


In [3]:
werrv=["https://werrv.ca/collections/automotive-rv?sort_by=best-selling&filter.v.availability=1","https://werrv.ca/collections/marine-hardware?sort_by=best-selling&filter.v.availability=1",
"https://werrv.ca/collections/sailing?sort_by=best-selling&filter.v.availability=1",
"https://werrv.ca/collections/sailing?sort_by=best-selling&filter.v.availability=1",
"https://werrv.ca/collections/electrical",
"https://werrv.ca/collections/camping?sort_by=best-selling&filter.v.availability=1",
"https://werrv.ca/collections/watersports?sort_by=best-selling&filter.v.availability=1"]
print(werrv)

['https://werrv.ca/collections/automotive-rv?sort_by=best-selling&filter.v.availability=1', 'https://werrv.ca/collections/marine-hardware?sort_by=best-selling&filter.v.availability=1', 'https://werrv.ca/collections/sailing?sort_by=best-selling&filter.v.availability=1', 'https://werrv.ca/collections/sailing?sort_by=best-selling&filter.v.availability=1', 'https://werrv.ca/collections/electrical', 'https://werrv.ca/collections/camping?sort_by=best-selling&filter.v.availability=1', 'https://werrv.ca/collections/watersports?sort_by=best-selling&filter.v.availability=1']


In [4]:
for i in werrv:
    print(i)

https://werrv.ca/collections/automotive-rv?sort_by=best-selling&filter.v.availability=1
https://werrv.ca/collections/marine-hardware?sort_by=best-selling&filter.v.availability=1
https://werrv.ca/collections/sailing?sort_by=best-selling&filter.v.availability=1
https://werrv.ca/collections/sailing?sort_by=best-selling&filter.v.availability=1
https://werrv.ca/collections/electrical
https://werrv.ca/collections/camping?sort_by=best-selling&filter.v.availability=1
https://werrv.ca/collections/watersports?sort_by=best-selling&filter.v.availability=1


In [5]:
import requests
from bs4 import BeautifulSoup
import pandas as pd

data = []


# Start URL
for i in werrv:
    url=i

    
    

    while url:
        
        response = requests.get(url)

    
        soup = BeautifulSoup(response.text, 'html.parser')

    
        products = soup.find_all("div", {"class": "product-item__info-inner"})

        for product in products:
            title = product.find("a", {"class": "product-item__title text--strong link"}).text
            price = product.find("div", {"class": "product-item__price-list price-list"}).text
            data.append([title,price])
    
    
        next_button = soup.find("a", {"class": "pagination__next link"})

    
        if next_button:
            url = "https://werrv.ca" + next_button['href']
        else:
        
            url = None

        
        
        


df = pd.DataFrame(data, columns=['Title', 'Price'])


df.to_excel("allproducts.xlsx", index=False)


In [6]:
# are the categories are the only unique products they have? or do they want that particular products to be ranked at top?
# do they want products from all collections too?
# whats the difference between Brands and all collection ? will they have a same produts are different?


In [7]:
# https://werrv.ca/collections
# create a loop for collecting all the products from this collection

# also there is a brand name in the bottom of the page


# using the below code we can take all the categories from the website

In [28]:
import requests
from bs4 import BeautifulSoup
import pandas as pd

# Start URL
url = "https://werrv.ca/collections"

#initialize empty list
data = []

while url:
    # Send a GET request to the URL
    response = requests.get(url)

    # Create a BeautifulSoup object to parse the HTML content
    soup = BeautifulSoup(response.text, 'html.parser')

    # Find all the products on the page
    products = soup.find_all("div", {"class": "grid__cell 1/2--tablet 1/3--lap-and-up"})

    for product in products:
        title = product.find("a", {"class": "collection-block-item collection-block-item--overlay"}).text
#         price = product.find("div", {"class": "product-item__price-list price-list"}).text
        data.append([title])
    
    # Find the next page button
    next_button = soup.find("a", {"class": "pagination__next link"})

    # If there is a next page button, update the URL
    if next_button:
        url = "https://werrv.ca" + next_button['href']
    else:
        # If there is no next page button, break out of the loop
        url = None
        
# Create a dataframe from the scraped data
df = pd.DataFrame(data, columns=['Title'])

# Export the dataframe to an excel sheet
df.to_csv("allcollection3.csv", index=False)


In [ ]:
https://werrv.ca/collections/accessories?sort_by=best-selling&filter.v.availability=1
https://werrv.ca/collections/anchoring-docking?sort_by=best-selling&filter.v.availability=1
https://werrv.ca/collections/anchoring-docking-anchoring-accessories?sort_by=best-selling&filter.v.availability=1


In [21]:
# then we gonna load that file to run in the for loop inorder to get all the product items from the list

In [131]:
import pandas as pd
df= pd.read_csv("allcollection3.csv")
df.head()


Title
0                                  \n\n\nAccessories
1                          \n\n\nAnchoring & Docking
2  \n\n\nAnchoring & Docking - Anchoring Accessories
3                \n\n\nAnchoring & Docking - Anchors
4         \n\n\nAnchoring & Docking - Bumpers/Guards

In [132]:
import pandas as pd
df = pd.read_csv('allcollection3.csv',header=0)
df.replace('\n\n\n', '', regex=True, inplace=True)
df.to_csv('cleaned1.csv', index=False)

In [133]:
df1= pd.read_csv("cleaned1.csv",header=0)
df1

Title
0                                    Accessories
1                            Anchoring & Docking
2    Anchoring & Docking - Anchoring Accessories
3                  Anchoring & Docking - Anchors
4           Anchoring & Docking - Bumpers/Guards
..                                           ...
629                       Winterizing - De-icers
630          Winterizing - Heaters/Dehumidifiers
631             Winterizing - Oil Change Systems
632         Winterizing - Water Flushing Systems
633                        Winterizing Boat Kits

[634 rows x 1 columns]

In [134]:
df1.set_index('Title', inplace=True)

In [145]:
column_name = 'Title'
column_data = df['Title'].tolist()
print(column_data)

['Accessories', 'Anchoring & Docking', 'Anchoring & Docking - Anchoring Accessories', 'Anchoring & Docking - Anchors', 'Anchoring & Docking - Bumpers/Guards', 'Anchoring & Docking - Buoys', 'Anchoring & Docking - Cleats', 'Anchoring & Docking - Dock Line', 'Anchoring & Docking - Docking Accessories', 'Anchoring & Docking - Fender Accessories', 'Anchoring & Docking - Fender Covers', 'Anchoring & Docking - Fenders', 'Anchoring & Docking - Ladders', 'Anchoring & Docking - Mooring Whips', 'Anchoring & Docking - Piling Caps', 'Anchoring & Docking - Rope & Chain', 'Anchoring & Docking - Windlass Accessories', 'Anchoring & Docking - Windlasses', 'Automotive/RV - Covers', 'Awnings & Shades', 'Awnings & Shades - Awning Parts & Accessories', 'Awnings & Shades - Awning Rooms', 'Awnings & Shades - Canopies & Shelters', 'Awnings & Shades - Replacement Fabrics', 'Awnings & Shades - RV Patio Awnings', 'Awnings & Shades - RV Slide-out Awnings', 'Awnings & Shades - Sun Blockers', 'Awnings & Shades - Wi

In [146]:
collection=[]
for i in column_data:
    j=i.lower()
    collection.append(j)

In [147]:
collection

['accessories',
 'anchoring & docking',
 'anchoring & docking - anchoring accessories',
 'anchoring & docking - anchors',
 'anchoring & docking - bumpers/guards',
 'anchoring & docking - buoys',
 'anchoring & docking - cleats',
 'anchoring & docking - dock line',
 'anchoring & docking - docking accessories',
 'anchoring & docking - fender accessories',
 'anchoring & docking - fender covers',
 'anchoring & docking - fenders',
 'anchoring & docking - ladders',
 'anchoring & docking - mooring whips',
 'anchoring & docking - piling caps',
 'anchoring & docking - rope & chain',
 'anchoring & docking - windlass accessories',
 'anchoring & docking - windlasses',
 'automotive/rv - covers',
 'awnings & shades',
 'awnings & shades - awning parts & accessories',
 'awnings & shades - awning rooms',
 'awnings & shades - canopies & shelters',
 'awnings & shades - replacement fabrics',
 'awnings & shades - rv patio awnings',
 'awnings & shades - rv slide-out awnings',
 'awnings & shades - sun blocker

In [148]:
for i in range(len(collection)):
    collection[i] = collection[i].replace("&", "-")

print(collection)

['accessories', 'anchoring - docking', 'anchoring - docking - anchoring accessories', 'anchoring - docking - anchors', 'anchoring - docking - bumpers/guards', 'anchoring - docking - buoys', 'anchoring - docking - cleats', 'anchoring - docking - dock line', 'anchoring - docking - docking accessories', 'anchoring - docking - fender accessories', 'anchoring - docking - fender covers', 'anchoring - docking - fenders', 'anchoring - docking - ladders', 'anchoring - docking - mooring whips', 'anchoring - docking - piling caps', 'anchoring - docking - rope - chain', 'anchoring - docking - windlass accessories', 'anchoring - docking - windlasses', 'automotive/rv - covers', 'awnings - shades', 'awnings - shades - awning parts - accessories', 'awnings - shades - awning rooms', 'awnings - shades - canopies - shelters', 'awnings - shades - replacement fabrics', 'awnings - shades - rv patio awnings', 'awnings - shades - rv slide-out awnings', 'awnings - shades - sun blockers', 'awnings - shades - wi

In [150]:
for i in range(len(collection)):
    collection[i] = collection[i].replace(",", "-")

print(collection)

['accessories', 'anchoring - docking', 'anchoring - docking - anchoring accessories', 'anchoring - docking - anchors', 'anchoring - docking - bumpers/guards', 'anchoring - docking - buoys', 'anchoring - docking - cleats', 'anchoring - docking - dock line', 'anchoring - docking - docking accessories', 'anchoring - docking - fender accessories', 'anchoring - docking - fender covers', 'anchoring - docking - fenders', 'anchoring - docking - ladders', 'anchoring - docking - mooring whips', 'anchoring - docking - piling caps', 'anchoring - docking - rope - chain', 'anchoring - docking - windlass accessories', 'anchoring - docking - windlasses', 'automotive/rv - covers', 'awnings - shades', 'awnings - shades - awning parts - accessories', 'awnings - shades - awning rooms', 'awnings - shades - canopies - shelters', 'awnings - shades - replacement fabrics', 'awnings - shades - rv patio awnings', 'awnings - shades - rv slide-out awnings', 'awnings - shades - sun blockers', 'awnings - shades - wi

In [151]:
collection

['accessories',
 'anchoring - docking',
 'anchoring - docking - anchoring accessories',
 'anchoring - docking - anchors',
 'anchoring - docking - bumpers/guards',
 'anchoring - docking - buoys',
 'anchoring - docking - cleats',
 'anchoring - docking - dock line',
 'anchoring - docking - docking accessories',
 'anchoring - docking - fender accessories',
 'anchoring - docking - fender covers',
 'anchoring - docking - fenders',
 'anchoring - docking - ladders',
 'anchoring - docking - mooring whips',
 'anchoring - docking - piling caps',
 'anchoring - docking - rope - chain',
 'anchoring - docking - windlass accessories',
 'anchoring - docking - windlasses',
 'automotive/rv - covers',
 'awnings - shades',
 'awnings - shades - awning parts - accessories',
 'awnings - shades - awning rooms',
 'awnings - shades - canopies - shelters',
 'awnings - shades - replacement fabrics',
 'awnings - shades - rv patio awnings',
 'awnings - shades - rv slide-out awnings',
 'awnings - shades - sun blocker

In [152]:
for i in range(len(collection)):
    collection[i] = collection[i].replace(" - ", "-")

print(collection)

['accessories', 'anchoring-docking', 'anchoring-docking-anchoring accessories', 'anchoring-docking-anchors', 'anchoring-docking-bumpers/guards', 'anchoring-docking-buoys', 'anchoring-docking-cleats', 'anchoring-docking-dock line', 'anchoring-docking-docking accessories', 'anchoring-docking-fender accessories', 'anchoring-docking-fender covers', 'anchoring-docking-fenders', 'anchoring-docking-ladders', 'anchoring-docking-mooring whips', 'anchoring-docking-piling caps', 'anchoring-docking-rope-chain', 'anchoring-docking-windlass accessories', 'anchoring-docking-windlasses', 'automotive/rv-covers', 'awnings-shades', 'awnings-shades-awning parts-accessories', 'awnings-shades-awning rooms', 'awnings-shades-canopies-shelters', 'awnings-shades-replacement fabrics', 'awnings-shades-rv patio awnings', 'awnings-shades-rv slide-out awnings', 'awnings-shades-sun blockers', 'awnings-shades-window-door awnings', 'bed- bath-laundry', 'bed- bath-laundry-bathroom sinks', 'bed- bath-laundry-bathroom sup

In [141]:
for i in range(len(collection)):
    collection[i] = collection[i].replace(" ", "-")

print(collection)

['accessories', 'anchoring-docking', 'anchoring-docking-anchoring-accessories', 'anchoring-docking-anchors', 'anchoring-docking-bumpers/guards', 'anchoring-docking-buoys', 'anchoring-docking-cleats', 'anchoring-docking-dock-line', 'anchoring-docking-docking-accessories', 'anchoring-docking-fender-accessories', 'anchoring-docking-fender-covers', 'anchoring-docking-fenders', 'anchoring-docking-ladders', 'anchoring-docking-mooring-whips', 'anchoring-docking-piling-caps', 'anchoring-docking-rope-chain', 'anchoring-docking-windlass-accessories', 'anchoring-docking-windlasses', 'automotive/rv-covers', 'awnings-shades', 'awnings-shades-awning-parts-accessories', 'awnings-shades-awning-rooms', 'awnings-shades-canopies-shelters', 'awnings-shades-replacement-fabrics', 'awnings-shades-rv-patio-awnings', 'awnings-shades-rv-slide-out-awnings', 'awnings-shades-sun-blockers', 'awnings-shades-window-door-awnings', 'bed--bath-laundry', 'bed--bath-laundry-bathroom-sinks', 'bed--bath-laundry-bathroom-sup

In [153]:
collection

['accessories',
 'anchoring-docking',
 'anchoring-docking-anchoring accessories',
 'anchoring-docking-anchors',
 'anchoring-docking-bumpers/guards',
 'anchoring-docking-buoys',
 'anchoring-docking-cleats',
 'anchoring-docking-dock line',
 'anchoring-docking-docking accessories',
 'anchoring-docking-fender accessories',
 'anchoring-docking-fender covers',
 'anchoring-docking-fenders',
 'anchoring-docking-ladders',
 'anchoring-docking-mooring whips',
 'anchoring-docking-piling caps',
 'anchoring-docking-rope-chain',
 'anchoring-docking-windlass accessories',
 'anchoring-docking-windlasses',
 'automotive/rv-covers',
 'awnings-shades',
 'awnings-shades-awning parts-accessories',
 'awnings-shades-awning rooms',
 'awnings-shades-canopies-shelters',
 'awnings-shades-replacement fabrics',
 'awnings-shades-rv patio awnings',
 'awnings-shades-rv slide-out awnings',
 'awnings-shades-sun blockers',
 'awnings-shades-window-door awnings',
 'bed- bath-laundry',
 'bed- bath-laundry-bathroom sinks',
 '

In [154]:
for i in range(len(collection)):
    collection[i] = collection[i].replace("- ", "-")

print(collection)

['accessories', 'anchoring-docking', 'anchoring-docking-anchoring accessories', 'anchoring-docking-anchors', 'anchoring-docking-bumpers/guards', 'anchoring-docking-buoys', 'anchoring-docking-cleats', 'anchoring-docking-dock line', 'anchoring-docking-docking accessories', 'anchoring-docking-fender accessories', 'anchoring-docking-fender covers', 'anchoring-docking-fenders', 'anchoring-docking-ladders', 'anchoring-docking-mooring whips', 'anchoring-docking-piling caps', 'anchoring-docking-rope-chain', 'anchoring-docking-windlass accessories', 'anchoring-docking-windlasses', 'automotive/rv-covers', 'awnings-shades', 'awnings-shades-awning parts-accessories', 'awnings-shades-awning rooms', 'awnings-shades-canopies-shelters', 'awnings-shades-replacement fabrics', 'awnings-shades-rv patio awnings', 'awnings-shades-rv slide-out awnings', 'awnings-shades-sun blockers', 'awnings-shades-window-door awnings', 'bed-bath-laundry', 'bed-bath-laundry-bathroom sinks', 'bed-bath-laundry-bathroom suppli

In [157]:
for i in range(len(collection)):
    collection[i] = collection[i].replace("/", "-")

print(collection)

['accessories', 'anchoring-docking', 'anchoring-docking-anchoring accessories', 'anchoring-docking-anchors', 'anchoring-docking-bumpers-guards', 'anchoring-docking-buoys', 'anchoring-docking-cleats', 'anchoring-docking-dock line', 'anchoring-docking-docking accessories', 'anchoring-docking-fender accessories', 'anchoring-docking-fender covers', 'anchoring-docking-fenders', 'anchoring-docking-ladders', 'anchoring-docking-mooring whips', 'anchoring-docking-piling caps', 'anchoring-docking-rope-chain', 'anchoring-docking-windlass accessories', 'anchoring-docking-windlasses', 'automotive-rv-covers', 'awnings-shades', 'awnings-shades-awning parts-accessories', 'awnings-shades-awning rooms', 'awnings-shades-canopies-shelters', 'awnings-shades-replacement fabrics', 'awnings-shades-rv patio awnings', 'awnings-shades-rv slide-out awnings', 'awnings-shades-sun blockers', 'awnings-shades-window-door awnings', 'bed-bath-laundry', 'bed-bath-laundry-bathroom sinks', 'bed-bath-laundry-bathroom suppli

In [158]:
collection

['accessories',
 'anchoring-docking',
 'anchoring-docking-anchoring accessories',
 'anchoring-docking-anchors',
 'anchoring-docking-bumpers-guards',
 'anchoring-docking-buoys',
 'anchoring-docking-cleats',
 'anchoring-docking-dock line',
 'anchoring-docking-docking accessories',
 'anchoring-docking-fender accessories',
 'anchoring-docking-fender covers',
 'anchoring-docking-fenders',
 'anchoring-docking-ladders',
 'anchoring-docking-mooring whips',
 'anchoring-docking-piling caps',
 'anchoring-docking-rope-chain',
 'anchoring-docking-windlass accessories',
 'anchoring-docking-windlasses',
 'automotive-rv-covers',
 'awnings-shades',
 'awnings-shades-awning parts-accessories',
 'awnings-shades-awning rooms',
 'awnings-shades-canopies-shelters',
 'awnings-shades-replacement fabrics',
 'awnings-shades-rv patio awnings',
 'awnings-shades-rv slide-out awnings',
 'awnings-shades-sun blockers',
 'awnings-shades-window-door awnings',
 'bed-bath-laundry',
 'bed-bath-laundry-bathroom sinks',
 'be

In [159]:
for i in range(len(collection)):
    collection[i] = collection[i].replace(" ", "-")

print(collection)

['accessories', 'anchoring-docking', 'anchoring-docking-anchoring-accessories', 'anchoring-docking-anchors', 'anchoring-docking-bumpers-guards', 'anchoring-docking-buoys', 'anchoring-docking-cleats', 'anchoring-docking-dock-line', 'anchoring-docking-docking-accessories', 'anchoring-docking-fender-accessories', 'anchoring-docking-fender-covers', 'anchoring-docking-fenders', 'anchoring-docking-ladders', 'anchoring-docking-mooring-whips', 'anchoring-docking-piling-caps', 'anchoring-docking-rope-chain', 'anchoring-docking-windlass-accessories', 'anchoring-docking-windlasses', 'automotive-rv-covers', 'awnings-shades', 'awnings-shades-awning-parts-accessories', 'awnings-shades-awning-rooms', 'awnings-shades-canopies-shelters', 'awnings-shades-replacement-fabrics', 'awnings-shades-rv-patio-awnings', 'awnings-shades-rv-slide-out-awnings', 'awnings-shades-sun-blockers', 'awnings-shades-window-door-awnings', 'bed-bath-laundry', 'bed-bath-laundry-bathroom-sinks', 'bed-bath-laundry-bathroom-suppli

In [160]:
collection

['accessories',
 'anchoring-docking',
 'anchoring-docking-anchoring-accessories',
 'anchoring-docking-anchors',
 'anchoring-docking-bumpers-guards',
 'anchoring-docking-buoys',
 'anchoring-docking-cleats',
 'anchoring-docking-dock-line',
 'anchoring-docking-docking-accessories',
 'anchoring-docking-fender-accessories',
 'anchoring-docking-fender-covers',
 'anchoring-docking-fenders',
 'anchoring-docking-ladders',
 'anchoring-docking-mooring-whips',
 'anchoring-docking-piling-caps',
 'anchoring-docking-rope-chain',
 'anchoring-docking-windlass-accessories',
 'anchoring-docking-windlasses',
 'automotive-rv-covers',
 'awnings-shades',
 'awnings-shades-awning-parts-accessories',
 'awnings-shades-awning-rooms',
 'awnings-shades-canopies-shelters',
 'awnings-shades-replacement-fabrics',
 'awnings-shades-rv-patio-awnings',
 'awnings-shades-rv-slide-out-awnings',
 'awnings-shades-sun-blockers',
 'awnings-shades-window-door-awnings',
 'bed-bath-laundry',
 'bed-bath-laundry-bathroom-sinks',
 'be

In [91]:
# for example we gonna create a list and try to fit the data inside the link
mani=['accessories']

In [96]:
for i in mani:
    jai=f"https://werrv.ca/collections/{i}?sort_by=best-selling&filter.v.availability=1"

In [2]:
# #we didnt get anything from this code need toi check the html

# import requests
# from bs4 import BeautifulSoup
# import pandas as pd

# kai = []


# # Start URL
# for i in collection:
#     mani=f"https://werrv.ca/collections/{i}?sort_by=best-selling&filter.v.availability=1"
#     url=mani

    
    

#     while url:
        
#         response = requests.get(url)

    
#         soup = BeautifulSoup(response.text, 'html.parser')

    
#         products = soup.find_all("div", {"class": "product-item__info-inner"})

#         for product in products:
#             title = product.find("a", {"class": "product-item__title text--strong link"}).text
#             price = product.find("div", {"class": "product-item__price-list price-list"}).text
#             data.append([title,price])
    
    
#         next_button = soup.find("a", {"class": "pagination__next link"})

    
#         if next_button:
#             url = "https://werrv.ca" + next_button['href']
#         else:
        
#             url = None

        
        
        


# df = pd.DataFrame(kai, columns=['Title', 'Price'])


# df.to_excel("categoryproducts.xlsx", index=False)


In [161]:
collection

['accessories',
 'anchoring-docking',
 'anchoring-docking-anchoring-accessories',
 'anchoring-docking-anchors',
 'anchoring-docking-bumpers-guards',
 'anchoring-docking-buoys',
 'anchoring-docking-cleats',
 'anchoring-docking-dock-line',
 'anchoring-docking-docking-accessories',
 'anchoring-docking-fender-accessories',
 'anchoring-docking-fender-covers',
 'anchoring-docking-fenders',
 'anchoring-docking-ladders',
 'anchoring-docking-mooring-whips',
 'anchoring-docking-piling-caps',
 'anchoring-docking-rope-chain',
 'anchoring-docking-windlass-accessories',
 'anchoring-docking-windlasses',
 'automotive-rv-covers',
 'awnings-shades',
 'awnings-shades-awning-parts-accessories',
 'awnings-shades-awning-rooms',
 'awnings-shades-canopies-shelters',
 'awnings-shades-replacement-fabrics',
 'awnings-shades-rv-patio-awnings',
 'awnings-shades-rv-slide-out-awnings',
 'awnings-shades-sun-blockers',
 'awnings-shades-window-door-awnings',
 'bed-bath-laundry',
 'bed-bath-laundry-bathroom-sinks',
 'be

In [ ]:
# we got data for this usin g only one category from the collections

import requests
from bs4 import BeautifulSoup
import pandas as pd

data = []


# Start URL
for i in mani:
    url=jai

    
    

    while url:
        
        response = requests.get(url)

    
        soup = BeautifulSoup(response.text, 'html.parser')

    
        products = soup.find_all("div", {"class": "product-item__info-inner"})

        for product in products:
            title = product.find("a", {"class": "product-item__title text--strong link"}).text
            price = product.find("div", {"class": "product-item__price-list price-list"}).text
            data.append([title,price])
    
    
        next_button = soup.find("a", {"class": "pagination__next link"})

    
        if next_button:
            url = "https://werrv.ca" + next_button['href']
        else:
        
            url = None

        
        
        


df = pd.DataFrame(data, columns=['Title', 'Price'])


df.to_excel("allproducts1.xlsx", index=False)
 

In [101]:
df

Title  \
0     iN-Command Tire Pressure Monitoring System - 4...   
1     Garmin 12" Protective Cover - Plastic [010-127...   
2       Sea Catch TR3 w/Safety Pin - 1/4" Shackle [TR3]   
3     Camco D-I-Y Boat Winterizer Engine Flushing Sy...   
4     Lunasea Tri/Anchor/Flash Fixture Switch [LLB-5...   
...                                                 ...   
2148  Harken Marine Grip Tape - 6 x 12" - Black - 6 ...   
2149  Harken Marine Grip Tape - 3 x 12" - Black - 8 ...   
2150   Camco TastePURE RV & Marine Water Filter [40645]   
2151                      Stainless Steel Turkey Hanger   
2152                   PBC 14" PIT BARREL COOKER JUNIOR   

                                            Price  
0     \nSale price$437.00\n\nRegular price$577.00  
1       \nSale price$43.00\n\nRegular price$44.00  
2                             \nSale price$784.00  
3      \nSale price$75.00\n\nRegular price$123.00  
4      \nSale price$94.00\n\nRegular price$105.00  
...                                           ...  
2148                           \nSale price$84.00  
2149                           \nSale price$64.00  
2150    \nSale price$29.00\n\nRegular price$44.00  
2151                           \nSale price$60.00  
2152                          \nSale price$734.00  

[2153 rows x 2 columns]

In [162]:
for i in collection:
    mani=f"https://werrv.ca/collections/{i}?sort_by=best-selling&filter.v.availability=1"
    url=mani

In [163]:
anu=[]
for i in collection:
    x= f"https://werrv.ca/collections/{i}?sort_by=best-selling&filter.v.availability=1"
    anu.append(x)

In [164]:
for i in anu:
    print(i)

https://werrv.ca/collections/accessories?sort_by=best-selling&filter.v.availability=1
https://werrv.ca/collections/anchoring-docking?sort_by=best-selling&filter.v.availability=1
https://werrv.ca/collections/anchoring-docking-anchoring-accessories?sort_by=best-selling&filter.v.availability=1
https://werrv.ca/collections/anchoring-docking-anchors?sort_by=best-selling&filter.v.availability=1
https://werrv.ca/collections/anchoring-docking-bumpers-guards?sort_by=best-selling&filter.v.availability=1
https://werrv.ca/collections/anchoring-docking-buoys?sort_by=best-selling&filter.v.availability=1
https://werrv.ca/collections/anchoring-docking-cleats?sort_by=best-selling&filter.v.availability=1
https://werrv.ca/collections/anchoring-docking-dock-line?sort_by=best-selling&filter.v.availability=1
https://werrv.ca/collections/anchoring-docking-docking-accessories?sort_by=best-selling&filter.v.availability=1
https://werrv.ca/collections/anchoring-docking-fender-accessories?sort_by=best-selling&fil

In [169]:
import requests
from bs4 import BeautifulSoup
import pandas as pd

vetha = []


for i in anu:
#     mani=f"https://werrv.ca/collections/{i}?sort_by=best-selling&filter.v.availability=1"
    url=i
    
    

    while url:
        
        response = requests.get(url)

    
        soup = BeautifulSoup(response.text, 'html.parser')

    
        products = soup.find_all("div", {"class": "product-item__info-inner"})

        for product in products:
            title = product.find("a", {"class": "product-item__title text--strong link"}).text
            price = product.find("div", {"class": "product-item__price-list price-list"}).text
            vetha.append([title,price])
    
    
        next_button = soup.find("a", {"class": "pagination__next link"})

    
        if next_button:
            url = "https://werrv.ca" + next_button['href']
        else:
        
            url = None

        
        
        


df = pd.DataFrame(vetha, columns=['Title', 'Price'])


df.to_excel("allcollectionprice.xlsx", index=False)
 

In [1]:
# we got the product list from the entire website
# next we need to use the same in google in order to get the first 2 links. and need to check the keywords and implement 
#in our website

In [58]:
import pandas as pd
main=pd.read_excel("allcollectionprice.xlsx")

In [59]:
main["Title"]

0        iN-Command Tire Pressure Monitoring System - 4...
1        Garmin 12" Protective Cover - Plastic [010-127...
2          Sea Catch TR3 w/Safety Pin - 1/4" Shackle [TR3]
3        Camco D-I-Y Boat Winterizer Engine Flushing Sy...
4        Lunasea Tri/Anchor/Flash Fixture Switch [LLB-5...
                               ...                        
44617    Camco Quick Turn Permanent Waterheater Bypass ...
44618             Panther Inboard HD Flush Buddy [75-5000]
44619    Whitecap Universal Motor Flusher w/Rectangular...
44620    Whitecap Pro Dual Feed Motor Flusher w/Rectang...
44621    Camco Blow Out Plug - Plastic - Screws Into Wa...
Name: Title, Length: 44622, dtype: object

In [60]:
example=main["Title"].head(10)
print(example)

0    iN-Command Tire Pressure Monitoring System - 4...
1    Garmin 12" Protective Cover - Plastic [010-127...
2      Sea Catch TR3 w/Safety Pin - 1/4" Shackle [TR3]
3    Camco D-I-Y Boat Winterizer Engine Flushing Sy...
4    Lunasea Tri/Anchor/Flash Fixture Switch [LLB-5...
5    Garmin Protective Cover f/ECHOMAP Plus/UHD 7" ...
6    Humminbird IDMK S10 In-Dash Mounting Kit f/SOL...
7         Flojet Locking Slide Clips - Pair [20408000]
8    Forespar Nash 207 Racing Signal Flag Label [10...
9           Taylor Made 1/2" D-Sponge Gasket 7' [1725]
Name: Title, dtype: object


In [61]:
for i in example:
    print(i)
    

iN-Command Tire Pressure Monitoring System - 4 Sensor  Repeater Package [NCTP100]
Garmin 12" Protective Cover - Plastic [010-12799-01]
Sea Catch TR3 w/Safety Pin - 1/4" Shackle [TR3]
Camco D-I-Y Boat Winterizer Engine Flushing System [65501]
Lunasea Tri/Anchor/Flash Fixture Switch [LLB-53SW-81-00]
Garmin Protective Cover f/ECHOMAP Plus/UHD 7" Units [010-13126-00]
Humminbird IDMK S10 In-Dash Mounting Kit f/SOLIX 10 [740173-1]
Flojet Locking Slide Clips - Pair [20408000]
Forespar Nash 207 Racing Signal Flag Label [100017]
Taylor Made 1/2" D-Sponge Gasket 7' [1725]


In [77]:
#we gonna do google search inside the python code : idea: gonna use all the title from the products which we collected
# and we gonna use that in this google search loop for seacrhing and finding the top rank websites and then gonna scrap the 
# required information from the website

In [62]:
# !pip install googlesearch
from googlesearch import search

In [63]:
# pip install googlesearch-python

In [64]:
from googlesearch import search
s=input("keywords: ")
p= search(s, num_results=10)
mani=[]
for i in p:
    mani.append(i)

print(mani)

keywords: python
['https://www.python.org/', 'https://www.python.org/', 'https://en.wikipedia.org/wiki/Python_(programming_language)', 'https://www.w3schools.com/python/', 'https://www.codecademy.com/catalog/language/python', 'https://marketplace.visualstudio.com/items?itemName=ms-python.python', 'https://github.com/python', 'https://www.coursera.org/specializations/python', 'https://www.datacamp.com/courses/intro-to-python-for-data-science', 'https://google.github.io/styleguide/pyguide.html']


In [24]:
# from googlesearch import search
# s=input("keywords: ")
# p= search(s, num_results=10)
# mani=[]
# for i in p:
#     mani.append(i)

# print(mani)

In [25]:
# # from google import google

# # Number of results per page
# num_page = 1

# # Search query
# query = "Python programming language"

# results = search(query, num_page)

# for result in results:
#     print(result.name)
#     print(result.link)
#     print("\n")

In [18]:
import pandas as pd
df1=pd.read_excel("allcollectionprice.xlsx")
mani=df1["Title"].head(5)



In [17]:
from termcolor import colored
print(colored('Hello, World!', 'red', 'on_yellow', ['bold']))

Hello, World!


In [67]:
from googlesearch import search
from termcolor import colored

# Search query
for i in mani:
    query=i
    result= search(query, num_results=5)
    print(colored(i, 'red', 'on_yellow', ['bold']))
    for j in result:
        print(j)

iN-Command Tire Pressure Monitoring System - 4 Sensor  Repeater Package [NCTP100]
https://www.amazon.ca/Command-NCTP100-Monitoring-Bluetooth-Enabled-Bluetooth/dp/B0848SHHSN
https://www.bestbuy.ca/en-ca/product/in-command-tire-pressure-monitoring-system-4-sensor-amp-repeater-package/14665350
https://recreation-outfitters.com/products/in-command-tire-pressure-monitoring-system-4-sensor-repeater-package-nctp100
https://asasupportdocs.blob.core.windows.net/docs/Owners%20Manual/NCTP100%20(iN-Command%20TPMS)%20User%20manual%20for%20MP_20190802.pdf
https://premium-supply.com/products/nctp100-in-command-tire-pressure-monitoring-system
https://shut-up-and-fish.com/in-command-tire-pressure-monitoring-system-4-sensor-repeater-package-nctp100/
https://tricitiestackle.com/products/in-command-tire-pressure-monitoring-system-4-sensor-repeater-package-nctp100
Garmin 12" Protective Cover - Plastic [010-12799-01]
https://werrv.ca/products/garmin-12-protective-cover-plastic-010-12799-01
https://tetronmar

In [9]:
# beacuse of the overload and traffic we couldnt able to use all the url from the website

# TAKEN FROM BEST BUY

In [ ]:
import requests
from bs4 import BeautifulSoup

url = "https://www.bestbuy.ca/en-ca/product/in-command-tire-pressure-monitoring-system-4-sensor-amp-repeater-package/14665350"

response = requests.get(url)
soup = BeautifulSoup(response.content, "html.parser")

# Extract the product title
product_title = soup.find("h1",{"class":"productName_2KoPa"}).text
print("Product Title:", product_title)

# Extract the product price
product_price = soup.find("span",{"class":"product-price"}).text
print("Product Price:", product_price)

product_description = soup.find("div",{"class":"productDescription_2WBlx"}).text
print("Product Price:", product_price)

In [26]:
string="Tire Pressure Monitoring System - 4 Sensor & Repeater Package. Monitor the air pressure and temperature of up to 22 tires in real-time with the iN•Command Tire Pressure Monitoring System. The waterproof sensors use Bluetooth technology to communicate important safety information right to your smart device with the iN•Command TPMS App. . The repeater extends the Bluetooth signal for longer vehicle (up to 100 Meters or 300+ Feet) applications when mounted between the smartphone and rear tires. . Just open the Apple or Android App to assign a sensor to each tire by scanning the QR code with your device's camera to receive audible alerts and notifications.Features: Works with RVs, Trailers, Passenger Vehicles, and More, Each Tire can be Monitored Individually for Pressure and Temperature (F° or C°), Ability to Monitor up to 22 Tires at Once w/ the iN-Command app, Lets you Access Tire Pressure and Temperature History, Free App is Available on Android and Apple Smartphones, Pressure monitoring range: 0 to 185 PSI.In the Box:Four (4) Bluetooth-Enabled Tire Pressure Monitoring Sensors w/ Locking Anti-Theft Bolts, Four (4) 3V - CR1632 Batteries, Bluetooth Repeater, Installation Wrench, Instruction Manual."

In [39]:
string

"Tire Pressure Monitoring System - 4 Sensor & Repeater Package. Monitor the air pressure and temperature of up to 22 tires in real-time with the iN•Command Tire Pressure Monitoring System. The waterproof sensors use Bluetooth technology to communicate important safety information right to your smart device with the iN•Command TPMS App. . The repeater extends the Bluetooth signal for longer vehicle (up to 100 Meters or 300+ Feet) applications when mounted between the smartphone and rear tires. . Just open the Apple or Android App to assign a sensor to each tire by scanning the QR code with your device's camera to receive audible alerts and notifications.Features: Works with RVs, Trailers, Passenger Vehicles, and More, Each Tire can be Monitored Individually for Pressure and Temperature (F° or C°), Ability to Monitor up to 22 Tires at Once w/ the iN-Command app, Lets you Access Tire Pressure and Temperature History, Free App is Available on Android and Apple Smartphones, Pressure monitor

In [40]:
len(string)

1215

In [4]:
import nltk
nltk.download('stopwords')
from nltk.corpus import stopwords
from nltk.tokenize import word_tokenize

# Your string
# string = "This is a sample string with some common words that will be removed."

# Tokenize the string
words = word_tokenize(string)

# Get a list of stop words
stop_words = set(stopwords.words('english'))

# Remove stop words from the list of words
filtered_words = [word for word in words if word.lower() not in stop_words]

[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\91887\AppData\Roaming\nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


NameError: name 'string' is not defined

In [42]:
filtered_words

['Tire',
 'Pressure',
 'Monitoring',
 'System',
 '-',
 '4',
 'Sensor',
 '&',
 'Repeater',
 'Package',
 '.',
 'Monitor',
 'air',
 'pressure',
 'temperature',
 '22',
 'tires',
 'real-time',
 'iN•Command',
 'Tire',
 'Pressure',
 'Monitoring',
 'System',
 '.',
 'waterproof',
 'sensors',
 'use',
 'Bluetooth',
 'technology',
 'communicate',
 'important',
 'safety',
 'information',
 'right',
 'smart',
 'device',
 'iN•Command',
 'TPMS',
 'App.',
 '.',
 'repeater',
 'extends',
 'Bluetooth',
 'signal',
 'longer',
 'vehicle',
 '(',
 '100',
 'Meters',
 '300+',
 'Feet',
 ')',
 'applications',
 'mounted',
 'smartphone',
 'rear',
 'tires.',
 '.',
 'open',
 'Apple',
 'Android',
 'App',
 'assign',
 'sensor',
 'tire',
 'scanning',
 'QR',
 'code',
 'device',
 "'s",
 'camera',
 'receive',
 'audible',
 'alerts',
 'notifications.Features',
 ':',
 'Works',
 'RVs',
 ',',
 'Trailers',
 ',',
 'Passenger',
 'Vehicles',
 ',',
 ',',
 'Tire',
 'Monitored',
 'Individually',
 'Pressure',
 'Temperature',
 '(',
 'F°',


In [43]:
import re
puncuation = re.compile(r'[-.,?!:&+;()$%|0-9]')
post=[]
for i in filtered_words:
    word= puncuation.sub("",i)
    if len(word)>0:
        post.append(word)

In [44]:
post

['Tire',
 'Pressure',
 'Monitoring',
 'System',
 'Sensor',
 'Repeater',
 'Package',
 'Monitor',
 'air',
 'pressure',
 'temperature',
 'tires',
 'realtime',
 'iN•Command',
 'Tire',
 'Pressure',
 'Monitoring',
 'System',
 'waterproof',
 'sensors',
 'use',
 'Bluetooth',
 'technology',
 'communicate',
 'important',
 'safety',
 'information',
 'right',
 'smart',
 'device',
 'iN•Command',
 'TPMS',
 'App',
 'repeater',
 'extends',
 'Bluetooth',
 'signal',
 'longer',
 'vehicle',
 'Meters',
 'Feet',
 'applications',
 'mounted',
 'smartphone',
 'rear',
 'tires',
 'open',
 'Apple',
 'Android',
 'App',
 'assign',
 'sensor',
 'tire',
 'scanning',
 'QR',
 'code',
 'device',
 "'s",
 'camera',
 'receive',
 'audible',
 'alerts',
 'notificationsFeatures',
 'Works',
 'RVs',
 'Trailers',
 'Passenger',
 'Vehicles',
 'Tire',
 'Monitored',
 'Individually',
 'Pressure',
 'Temperature',
 'F°',
 'C°',
 'Ability',
 'Monitor',
 'Tires',
 'w/',
 'iNCommand',
 'app',
 'Lets',
 'Access',
 'Tire',
 'Pressure',
 'Temp

In [45]:
from nltk.probability import FreqDist
fdist= FreqDist()

In [46]:
for words in post:
    fdist[words.lower()]+=1
fdist 

FreqDist({'pressure': 7, 'tire': 6, 'monitoring': 4, 'app': 4, 'repeater': 3, 'temperature': 3, 'tires': 3, 'bluetooth': 3, 'system': 2, 'sensor': 2, ...})

# TAKEN FROM WERRV

In [47]:
string1="Monitor the air pressure and temperature of up to 22 tires in real-time with the iN•Command® Tire Pressure Monitoring System. The waterproof sensors use Bluetooth technology to communicate important safety information right to your smart device with the iN•Command TPMS App.The repeater extends the Bluetooth signal for longer vehicle (up to 100 Meters or 300+ Feet) applications when mounted between the smartphone and rear tires.Just open the Apple or Android App to assign a sensor to each tire by scanning the QR code with your device's camera to receive audible alerts and notifications."

In [55]:
string1

"Monitor the air pressure and temperature of up to 22 tires in real-time with the iN•Command® Tire Pressure Monitoring System. The waterproof sensors use Bluetooth technology to communicate important safety information right to your smart device with the iN•Command TPMS App.The repeater extends the Bluetooth signal for longer vehicle (up to 100 Meters or 300+ Feet) applications when mounted between the smartphone and rear tires.Just open the Apple or Android App to assign a sensor to each tire by scanning the QR code with your device's camera to receive audible alerts and notifications."

In [56]:
len(string1)

592

In [57]:
words = word_tokenize(string1)

# Get a list of stop words
stop_words = set(stopwords.words('english'))

# Remove stop words from the list of words
filtered_words1 = [word for word in words if word.lower() not in stop_words]

In [23]:
filtered_words1

['Monitor',
 'air',
 'pressure',
 'temperature',
 '22',
 'tires',
 'real-time',
 'iN•Command®',
 'Tire',
 'Pressure',
 'Monitoring',
 'System',
 '.',
 'waterproof',
 'sensors',
 'use',
 'Bluetooth',
 'technology',
 'communicate',
 'important',
 'safety',
 'information',
 'right',
 'smart',
 'device',
 'iN•Command',
 'TPMS',
 'App.The',
 'repeater',
 'extends',
 'Bluetooth',
 'signal',
 'longer',
 'vehicle',
 '(',
 '100',
 'Meters',
 '300+',
 'Feet',
 ')',
 'applications',
 'mounted',
 'smartphone',
 'rear',
 'tires.Just',
 'open',
 'Apple',
 'Android',
 'App',
 'assign',
 'sensor',
 'tire',
 'scanning',
 'QR',
 'code',
 'device',
 "'s",
 'camera',
 'receive',
 'audible',
 'alerts',
 'notifications',
 '.']

In [58]:
import re
puncuation = re.compile(r'[-.,?!:&+;()$%|0-9]')
post1=[]
for i in filtered_words1:
    word= puncuation.sub("",i)
    if len(word)>0:
        post1.append(word)

In [59]:
post1

['Monitor',
 'air',
 'pressure',
 'temperature',
 'tires',
 'realtime',
 'iN•Command®',
 'Tire',
 'Pressure',
 'Monitoring',
 'System',
 'waterproof',
 'sensors',
 'use',
 'Bluetooth',
 'technology',
 'communicate',
 'important',
 'safety',
 'information',
 'right',
 'smart',
 'device',
 'iN•Command',
 'TPMS',
 'AppThe',
 'repeater',
 'extends',
 'Bluetooth',
 'signal',
 'longer',
 'vehicle',
 'Meters',
 'Feet',
 'applications',
 'mounted',
 'smartphone',
 'rear',
 'tiresJust',
 'open',
 'Apple',
 'Android',
 'App',
 'assign',
 'sensor',
 'tire',
 'scanning',
 'QR',
 'code',
 'device',
 "'s",
 'camera',
 'receive',
 'audible',
 'alerts',
 'notifications']

In [10]:
from nltk.probability import FreqDist
fdist= FreqDist()

In [61]:
for words in post1:
    fdist[words.lower()]+=1
fdist 

FreqDist({'pressure': 2, 'tire': 2, 'bluetooth': 2, 'device': 2, 'monitor': 1, 'air': 1, 'temperature': 1, 'tires': 1, 'realtime': 1, 'in•command®': 1, ...})

# Product name: 

## iN-Command Tire Pressure Monitoring System - 4 Sensor  Repeater Package [NCTP100]

## Taken from BestBuy

### Length of the description=1215

### FreqDist({'pressure': 7, 'tire': 6, 'monitoring': 4, 'app': 4, 'repeater': 3, 'temperature': 3, 'tires': 3, 'bluetooth': 3, 'system': 2, 'sensor': 2, ...})

## Taken from WERRV


### Length of the description= 592

### FreqDist({'pressure': 2, 'tire': 2, 'bluetooth': 2, 'device': 2, 'monitor': 1, 'air': 1, 'temperature': 1, 'tires': 1, 'realtime': 1, 'in•command®': 1, ...})

## after implementing KEYWORDS

# FreqDist({'tire': 8, 'pressure': 8, 'monitoring': 6, 'system': 6, 'repeater': 5, 'tires': 5, 'package': 4, 'sensors': 4, "'re": 4, 'readings': 4, ...})

In [1]:
# we need to automate the content creation using the keyword which we scrapped from the website

In [1]:
# using chatgpt
string3="The iN-Command Tire Pressure Monitoring System - 4 Sensor Repeater Package is a cutting-edge tire pressure monitoring system that allows you to keep a close eye on the pressure of your tires. This package includes four sensors and a repeater, giving you complete coverage of your tires and ensuring that you're always aware of any changes in pressure.The system uses state-of-the-art Bluetooth technology to communicate with a companion app on your smartphone. With the app, you can monitor the pressure and temperature of each tire in real-time, and receive notifications if there are any changes that warrant your attention.In addition to monitoring tire pressure, the iN-Command Tire Pressure Monitoring System also provides temperature readings, which can be a valuable tool for understanding the health of your tires and ensuring that they're running at their optimal temperature. The sensors are easy to install and provide accurate and reliable readings, so you can have peace of mind when you're on the road.The repeater included in this package helps to extend the range of the sensors, ensuring that you can receive accurate readings even if you're driving long distances. The repeater is easy to install and seamlessly integrates with the sensors, providing you with a comprehensive tire monitoring solution.Overall, the iN-Command Tire Pressure Monitoring System - 4 Sensor Repeater Package is a must-have for anyone who wants to ensure the safety and longevity of their tires. With its accurate readings, real-time notifications, and easy-to-use app, this system is the perfect way to stay on top of your tire pressure and ensure that your tires are always in the best condition possible."

In [2]:
len(string3)

1701

In [6]:
words = word_tokenize(string3)

# Get a list of stop words
stop_words2 = set(stopwords.words('english'))

# Remove stop words from the list of words
filtered_words2 = [word for word in words if word.lower() not in stop_words2]

In [7]:
filtered_words2

['iN-Command',
 'Tire',
 'Pressure',
 'Monitoring',
 'System',
 '-',
 '4',
 'Sensor',
 'Repeater',
 'Package',
 'cutting-edge',
 'tire',
 'pressure',
 'monitoring',
 'system',
 'allows',
 'keep',
 'close',
 'eye',
 'pressure',
 'tires',
 '.',
 'package',
 'includes',
 'four',
 'sensors',
 'repeater',
 ',',
 'giving',
 'complete',
 'coverage',
 'tires',
 'ensuring',
 "'re",
 'always',
 'aware',
 'changes',
 'pressure.The',
 'system',
 'uses',
 'state-of-the-art',
 'Bluetooth',
 'technology',
 'communicate',
 'companion',
 'app',
 'smartphone',
 '.',
 'app',
 ',',
 'monitor',
 'pressure',
 'temperature',
 'tire',
 'real-time',
 ',',
 'receive',
 'notifications',
 'changes',
 'warrant',
 'attention.In',
 'addition',
 'monitoring',
 'tire',
 'pressure',
 ',',
 'iN-Command',
 'Tire',
 'Pressure',
 'Monitoring',
 'System',
 'also',
 'provides',
 'temperature',
 'readings',
 ',',
 'valuable',
 'tool',
 'understanding',
 'health',
 'tires',
 'ensuring',
 "'re",
 'running',
 'optimal',
 'temper

In [8]:
import re
puncuation = re.compile(r'[-.,?!:&+;()$%|0-9]')
post1=[]
for i in filtered_words2:
    word= puncuation.sub("",i)
    if len(word)>0:
        post1.append(word)

In [11]:
for words in post1:
    fdist[words.lower()]+=1
fdist 

FreqDist({'tire': 8, 'pressure': 8, 'monitoring': 6, 'system': 6, 'repeater': 5, 'tires': 5, 'package': 4, 'sensors': 4, "'re": 4, 'readings': 4, ...})

In [ ]:
machine learning model
transfer model
bert, bart model...


In [5]:

import nltk
nltk.download('stopwords')
from nltk.corpus import stopwords
from nltk.tokenize import word_tokenize

[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\91887\AppData\Roaming\nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


In [10]:
from nltk.probability import FreqDist
fdist= FreqDist()

In [12]:
pip install openai

     ---------------------------------------- 55.6/55.6 kB 2.8 MB/s eta 0:00:00
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Installing backend dependencies: started
  Installing backend dependencies: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
  Created wheel for openai: filename=openai-0.26.4-py3-none-any.whl size=67767 sha256=9fd6750bae9990a9dae799d1b499e9d5b00ee7b53de44a10ba8e8522d16ef3a8
  Stored in directory: c:\users\91887\appdata\local\pip\cache\wheels\38\1a\4d\e4f9253b884e9d845680b2cff739b3ea07a7d2a7901de942a8
Successfully built openai
Note: you may need to restart the kernel to use updated packages.


In [34]:
import openai

# Set up the API client with your API key
openai.api_key = "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"

# Define the prompt for the content you want to generate
prompt = "write a product description with atleast 1500 words for the iN-Command Tire Pressure Monitoring System - 4 Sensor Repeater Package, using the following keywords and mentioned frequency pressure': 7, 'tire': 6, 'monitoring': 4, 'app': 4, 'repeater': 3, 'temperature': 3, 'tires': 3, 'bluetooth': 3, 'system': 2, 'sensor': 2 "

# Generate the content using the GPT-3 language model
response = openai.Completion.create(
    engine="text-davinci-002",
    prompt=prompt,
    max_tokens=2000,
    n=1,
    stop=None,
    temperature=0.5,
)

# Print the generated content
print(response["choices"][0]["text"])




The iN-Command Tire Pressure Monitoring System - 4 Sensor Repeater Package is a great way to keep an eye on your tire pressure. This system includes four sensors that monitor the pressure in your tires and send the information to a repeater. The repeater then sends the information to an app on your phone via Bluetooth. The app will show you the pressure in your tires, as well as the temperature. You can also set up the app to send you alerts if the pressure in your tires gets too low.


In [12]:
import openai

# Set up the API client with your API key
openai.api_key = "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"

# Define the prompt for the content you want to generate
prompt = "write a product description of the iN-Command Tire Pressure Monitoring System - 4 Sensor Repeater Package, mentioning the keywords 'pressure': 7 times, 'tire': 6 times, 'monitoring': 4 times, 'app': 4 times, 'repeater': 3 times, 'temperature': 3 times, 'tires': 3 times, 'bluetooth': 3 times, 'system': 2 times, 'sensor': 2 times with atleast 1500 words."

# Generate the content using the GPT-3 language model
response = openai.Completion.create(
    engine="text-davinci-002",
    prompt=prompt,
    max_tokens=1024,
    n=1,
    stop=None,
    temperature=0.5,
)

# Print the generated content
mani= response["choices"][0]["text"]


In [39]:
mani

'\n\nThe iN-Command Tire Pressure Monitoring System - 4 Sensor Repeater Package is a complete tire pressure monitoring system that includes four sensors and a repeater. The sensors attach to the tires and monitor the pressure and temperature of the tires. The repeater is placed in the vehicle and receives the signal from the sensors. The repeater then transmits the signal to the iN-Command app via Bluetooth. The app displays the pressure and temperature of the tires in real time. The system also includes a low pressure alarm that will notify the driver if the pressure in any of the tires is low.'

In [13]:
import openai

# Set up the API client with your API key
openai.api_key = "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"

# Define the prompt for the content you want to generate
prompt = "write a product description of the iN-Command Tire Pressure Monitoring System - 4 Sensor Repeater Package using the following example as a guide: \n\nExample: Monitor the air pressure and temperature of up to 22 tires in real-time with the iN•Command® Tire Pressure Monitoring System. The waterproof sensors use Bluetooth technology to communicate important safety information right to your smart device with the iN•Command TPMS App.The repeater extends the Bluetooth signal for longer vehicle (up to 100 Meters or 300+ Feet) applications when mounted between the smartphone and rear tires.Just open the Apple or Android App to assign a sensor to each tire by scanning the QR code with your device's camera to receive audible alerts and notifications."

# Generate the content using the GPT-3 language model
response = openai.Completion.create(
    engine="text-davinci-002",
    prompt=prompt,
    max_tokens=1024,
    n=1,
    stop=None,
    temperature=0.5,
)

# Print the generated content
print(response["choices"][0]["text"])





The iN-Command Tire Pressure Monitoring System lets you monitor the air pressure and temperature of up to 22 tires in real-time. The waterproof sensors use Bluetooth technology to communicate important safety information right to your smart device with the iN•Command TPMS App. The repeater extends the Bluetooth signal for longer vehicle (up to 100 Meters or 300+ Feet) applications when mounted between the smartphone and rear tires. Just open the Apple or Android App to assign a sensor to each tire by scanning the QR code with your device's camera to receive audible alerts and notifications.


In [41]:
mani

'\n\nThe iN-Command Tire Pressure Monitoring System - 4 Sensor Repeater Package is a complete tire pressure monitoring system that includes four sensors and a repeater. The sensors attach to the tires and monitor the pressure and temperature of the tires. The repeater is placed in the vehicle and receives the signal from the sensors. The repeater then transmits the signal to the iN-Command app via Bluetooth. The app displays the pressure and temperature of the tires in real time. The system also includes a low pressure alarm that will notify the driver if the pressure in any of the tires is low.'

In [48]:
string3=mani

In [49]:
words = word_tokenize(string3)

# Get a list of stop words
stop_words2 = set(stopwords.words('english'))

# Remove stop words from the list of words
filtered_words2 = [word for word in words if word.lower() not in stop_words2]

In [51]:
import re
puncuation = re.compile(r'[-.,?!:&+;()$%|/0-9]')
post5=[]
for i in filtered_words2:
    word= puncuation.sub("",i)
    if len(word)>0:
        post5.append(word)

In [52]:
for words in post5:
    fdist[words.lower()]+=1
fdist 

FreqDist({'pressure': 20, 'repeater': 13, 'tires': 13, 'tire': 12, 'system': 12, 'monitoring': 10, 'sensors': 10, 'incommand': 7, 'app': 7, 'temperature': 7, ...})

In [1]:
import openai

In [24]:
fdist=({'pressure': 20, 'repeater': 13, 'tires': 13, 'tire': 12, 'system': 12, 'monitoring': 10, 'sensors': 10, 'incommand': 7, 'app': 7, 'temperature': 7})

In [20]:
mani=df1["Title"].head(5)
mani

0    iN-Command Tire Pressure Monitoring System - 4...
1    Garmin 12" Protective Cover - Plastic [010-127...
2      Sea Catch TR3 w/Safety Pin - 1/4" Shackle [TR3]
3    Camco D-I-Y Boat Winterizer Engine Flushing Sy...
4    Lunasea Tri/Anchor/Flash Fixture Switch [LLB-5...
Name: Title, dtype: object

In [25]:
import openai

# Set up the API client with your API key
openai.api_key = "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"

for i in mani:
# Define the prompt for the content you want to generate
    prompt = f"write a product description for {i} with 1000 words using {fdist } this keywords and frquency"
    
# Generate the content using the GPT-3 language model
    response = openai.Completion.create(
        engine="text-davinci-002",
        prompt=prompt,
        max_tokens=2000,
        n=1,
        stop=None,
        temperature=0.5,
        )

# Print the generated content
    print(colored(i, 'red', 'on_yellow', ['bold']))    
    print(response["choices"][0]["text"])

iN-Command Tire Pressure Monitoring System - 4 Sensor  Repeater Package [NCTP100]


Looking for a way to keep an eye on your tire pressure? Check out the iN-Command Tire Pressure Monitoring System. This system comes with 4 sensors and a repeater so you can keep an eye on your tire pressure from inside your vehicle. You can also download the iN-Command app to get real-time alerts on your phone.
Garmin 12" Protective Cover - Plastic [010-12799-01]


Looking for a way to protect your Garmin 12" GPS unit? Look no further than the Garmin 12" Protective Cover - Plastic. This durable cover is made of tough plastic and is designed to help keep your GPS unit safe from the elements. The cover also features a built-in repeater for added protection from the elements.
Sea Catch TR3 w/Safety Pin - 1/4" Shackle [TR3]


Looking for a reliable and safe way to secure your boat? Look no further than the Sea Catch TR3 w/Safety Pin. This product is made with a strong and durable material that can withstand

In [28]:
from googlesearch import search
from termcolor import colored

In [29]:
import pandas as pd
df1=pd.read_excel("allcollectionprice.xlsx")
mani=df1["Title"].head(5)


In [1]:
import requests

url = "https://api.dataforseo.com/v3/serp/google/organic/live/advanced"

payload="[{\"keyword\":\"pressure\", \"location_code\":2124, \"language_code\":\"en\", \"device\":\"desktop\", \"os\":\"windows\", \"depth\":100, \"search_param\":\"https://www.bestbuy.ca/en-ca/product/in-command-tire-pressure-monitoring-system-4-sensor-amp-repeater-package/14665350\"}]"
headers = {
    'Authorization': 'Basic bG9naW46cGFzc3dvcmQ=',
    'Content-Type': 'application/json'
}

response = requests.request("POST", url, headers=headers, data=payload)

print(response.text)

{
    "version": "0.1.20221214",
    "status_code": 40100,
    "status_message": "You are not authorized to access this resource. See your login details here: https:\/\/app.dataforseo.com\/api-dashboard .",
    "time": "0 sec.",
    "cost": 0,
    "tasks_count": 0,
    "tasks_error": 0,
    "tasks": null
}


In [5]:
import openai

# Set up the API client with your API key
openai.api_key = "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"

# Define the prompt for the content you want to generate
prompt = "give me the content description for in command tire with 1000 words"

# Generate the content using the GPT-3 language model
response = openai.Completion.create(
    engine="text-davinci-002",
    prompt=prompt,
    max_tokens=1918,
    n=1,
    stop=None,
    temperature=0.5,
)

# Print the generated content
print(response["choices"][0]["text"])



In command tires are designed to provide drivers with a safe and reliable tire option for their vehicles. The tires are made with a special tread compound that helps to grip the road and provide traction in all types of weather conditions. The tires are also designed with a reinforced sidewall that helps to protect against punctures and other road hazards.


In [6]:
response

<OpenAIObject text_completion id=cmpl-6iAXq3zpftLvWm8ClDgPvBpbIxwWj at 0x2055be73be8> JSON: {
  "choices": [
    {
      "finish_reason": "stop",
      "index": 0,
      "logprobs": null,
      "text": "\n\nIn command tires are designed to provide drivers with a safe and reliable tire option for their vehicles. The tires are made with a special tread compound that helps to grip the road and provide traction in all types of weather conditions. The tires are also designed with a reinforced sidewall that helps to protect against punctures and other road hazards."
    }
  ],
  "created": 1675984986,
  "id": "cmpl-6iAXq3zpftLvWm8ClDgPvBpbIxwWj",
  "model": "text-davinci-002",
  "object": "text_completion",
  "usage": {
    "completion_tokens": 68,
    "prompt_tokens": 12,
    "total_tokens": 80
  }
}

In [4]:
import requests
from bs4 import BeautifulSoup

url = "https://serpapi.com/"

response = requests.get(url)
soup = BeautifulSoup(response.content, "html.parser")

# Extract the product title
jason_file = soup.find("h1",{"id":"results-json"}).json
print("jason file:", jason_file)



AttributeError: 'NoneType' object has no attribute 'json'